### PR TITLE
#164327420 Get red-flag records

### PR DESCRIPTION
**Pull request by AlvinMugambi**
**What does this PR do?**
Adds the Read in the CRUD for the red-flag incidents

**Tasks to be completed**
- write tests for the get red-flag endpoints
- add the get all records endpoint
- add the get one record endpoint

**How can this be manually tested**
Step 1

Navigate to your working directory, create and activate the virtual environment

```$ virtualenv -p python .venv```

```$ source .venv/bin/activate```

Clone the repository

```$ git clone https://github.com/ann-mukundi/ireporter.git```

Switch to the ireporter directory

```$ cd ireporter/```

Switch to testing branch

```$ git checkout ft-get-redflags-164327420```

Install project dependencies

```$ pip install -r api/requirements.txt```

Run the database migrations

```$ python api/manage.py makemigrations```
```$ python api/manage.py migrate```

Step 2
**Running the application**

```$ python api/manage.py runserver```

Step 3
**Login user**
Signup and activate user (if no user exists)
```<your-localhost>/api/auth/signup```

Login user (after activation)
```<your-localhost>/api/auth/login```

Step 4
**Add redflag**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/redflags/```

Step 5
**Get all redflags**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/redflags/```

Step 6
**Get one redflag**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/redflags/<id>``` where ```id``` is the specific red-flag id

**OR**

**Get one redflag**
Use the token provided after login for Authorization and the endpoint
```<your-localhost>/api/redflags/``` then click on the ```url``` in the one of the redflag details

What are the relevant PT stories?

[#164327420](https://www.pivotaltracker.com/story/show/164327420)

**Testing**

$ python manage.py test
**Prerequisites**

Python, Git, Virtualenv

**Screenshots**

**Get all redflags**
get all

**Get one redflag**
get 0ne

**Redflag not found**
red-flag not found

### CODE REVIEW

**Comment by ElishaMisoi** [comment]
Would be great to have a screenshot in the PR. Other than that, the code looks good to me 👍🏽

**Review by Siffersari — CHANGES_REQUESTED** [review]
You could add an extra line at the end of the serializer, views, and tests to comply with PEP8 Standards.

You could include some screenshots so that it easier to steal a glance.

Good job building out the endpoint.

File: api/incidents/views.py [hunk 1]
```diff
@@ -54,3 +54,32 @@ def create_redflag_record(self, request):
     def create(self, request):
         request.data['createdBy'] = request.user.id
         return self.validate_record(request)
+
+    def list(self, request):
```
Comment (meshack-mbuvi) [inline api/incidents/views.py:88]: combine the methods `list` and `retrieve` into one method

**Review by meshack-mbuvi — CHANGES_REQUESTED** [review]

**Comment by meshack-mbuvi** [comment]
@AlvinMugambi Your PR does not have screenshots. why so?

**Comment by AlvinMugambi** [comment]
> @AlvinMugambi Your PR does not have screenshots. why so?

Resolved.

**Comment by AlvinMugambi** [comment]
> You could add an extra line at the end of the serializer, views, and tests to comply with PEP8 Standards.
> 
> You could include some screenshots so that it easier to steal a glance.
> 
> Good job building out the endpoint.

Resolved. Seems develop also had no space at the end of the scripts. This has brought about merge conflicts

**Comment by AlvinMugambi** [comment]
> Would be great to have a screenshot in the PR. Other than that, the code looks good to me 👍🏽

Done.

File: api/incidents/views.py (hunk 1 above)
Comment (AlvinMugambi) [reply api/incidents/views.py:88]: For the view set we are using i.e the modelViewSet, we have to separate the two in order to override the default return messages.

**Pull request merged**